### PR TITLE
Expand test coverage with name-filter-fn.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR
- Add some unit tests for named-filter-fn == nil as well as non-null.
- 
- 

## Why are we making these changes?
- We test it == (constantly true), but name-filter-str->name-filter-fn only sets it to a regexp matcher closure or nil. 
- This exercises the special case nil in the query logic which wouldn't be exercised with the 'constantly true' existing implementation.

### CODE REVIEW

File: scheduler/test/cook/test/mesos/util.clj [hunk 1]
```diff
@@ -393,21 +393,50 @@
             _ (Thread/sleep 5)
             end-time (Date.)
             states [(name state)]
-            name (constantly true)]
+            namet (constantly true)
+            namef (constantly false)
```
Comment (dposada) [inline scheduler/test/cook/test/mesos/util.clj]: Maybe `match-any-name` and `match-no-name`?

File: scheduler/test/cook/test/mesos/util.clj [hunk 2]
```diff
@@ -393,21 +393,50 @@
...
+            ]
```
Comment (dposada) [inline scheduler/test/cook/test/mesos/util.clj]: The close bracket should be on the previous line

File: scheduler/test/cook/test/mesos/util.clj (hunk 1 above)
Comment (scrosby) [reply scheduler/test/cook/test/mesos/util.clj]: Renamed to match-any-name-fn and match-no-name-fn.

**Comment by dposada** [comment]
Needs rebase

File: scheduler/test/cook/test/mesos/util.clj (hunk 2 above)
Comment (dposada) [reply scheduler/test/cook/test/mesos/util.clj]: @scrosby Thoughts on this?

Comment (dposada) [reply scheduler/test/cook/test/mesos/util.clj]: @scrosby Thoughts on this?

Comment (scrosby) [reply scheduler/test/cook/test/mesos/util.clj]: Fixed in recent push.
